raspberryPI3/ros2_ws/src/car_control/src/car_control_node.cpp

```diff
@@ -148,15 +157,20 @@ class car_control : public rclcpp::Node {
 
             //Autonomous Mode
             } else if (mode==1){
-                
-                updateSpeedCmd();
-                
-                RCLCPP_DEBUG(this->get_logger(), "leftSpeedCmd : %f", leftSpeedCmd);
-                RCLCPP_DEBUG(this->get_logger(), "rightSpeedCmd : %f", rightSpeedCmd);
-
-                autonomousPropulsionCmd(rightSpeedCmd, rightRearPwmCmd);
-                autonomousPropulsionCmd(leftSpeedCmd, leftRearPwmCmd);
-                
+                if (!this->obstacle_detected){
```

Comment:
In general we should avoid doing that: `if (!myBool)`, except when doing error handling.
The reason why is that you can use the `else` branch for this body code and vice versa, and it is not natural to read the `!` or negative conditions. We may not see it, or even worse forget it.

Usually, electronics people often do that, so it is ok in our project, but if you work with informatics guys, they may become crazy because of this.